lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -0,0 +1,157 @@
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class GcsFileSystemConfig
+{
+    private DataSize readBlockSize = DataSize.of(4, DataSize.Unit.MEGABYTE);
+    private DataSize writeBlockSize = DataSize.of(4, DataSize.Unit.MEGABYTE);
+    private int pageSize = 100;
+    private int batchSize = 100;
+
+    private String projectId;
+
+    private boolean useGcsAccessToken;
+    private String jsonKey;
+    private String jsonKeyFilePath;
+
+    @NotNull
+    public DataSize getReadBlockSize()
+    {
+        return readBlockSize;
+    }
+
+    @Config("gcs.read-block-size")
+    public GcsFileSystemConfig setReadBlockSize(DataSize readBlockSize)
+    {
+        this.readBlockSize = readBlockSize;
+        return this;
+    }
+
+    @NotNull
+    public DataSize getWriteBlockSize()
+    {
+        return writeBlockSize;
+    }
+
+    @Config("gcs.write-block-size")
+    public GcsFileSystemConfig setWriteBlockSize(DataSize writeBlockSize)
+    {
+        this.writeBlockSize = writeBlockSize;
+        return this;
+    }
+
+    @Min(1)
+    public int getPageSize()
+    {
+        return pageSize;
+    }
+
+    @Config("gcs.page-size")
+    public GcsFileSystemConfig setPageSize(int pageSize)
+    {
+        this.pageSize = pageSize;
+        return this;
+    }
+
+    @Min(1)
+    public int getBatchSize()
+    {
+        return batchSize;
+    }
+
+    @Config("gcs.batch-size")
+    public GcsFileSystemConfig setBatchSize(int batchSize)
+    {
+        this.batchSize = batchSize;
+        return this;
+    }
+
+    @Nullable
+    public String getProjectId()
+    {
+        return projectId;
+    }
+
+    @Config("gcs.projectId")
+    public GcsFileSystemConfig setProjectId(String projectId)
+    {
+        this.projectId = projectId;
+        return this;
+    }
+
+    public boolean isUseGcsAccessToken()
+    {
+        return useGcsAccessToken;
+    }
+
+    @Config("gcs.use-access-token")
+    public GcsFileSystemConfig setUseGcsAccessToken(boolean useGcsAccessToken)
+    {
+        this.useGcsAccessToken = useGcsAccessToken;
+        return this;
+    }
+
+    @Nullable
+    public String getJsonKey()
+    {
+        return jsonKey;
+    }
+
+    @Config("gcs.json-key")
+    @ConfigSecuritySensitive
+    public GcsFileSystemConfig setJsonKey(String jsonKey)
+    {
+        this.jsonKey = jsonKey;
+        return this;
+    }
+
+    @Nullable
+    @FileExists
+    public String getJsonKeyFilePath()
+    {
+        return jsonKeyFilePath;
+    }
+
+    @Config("gcs.json-key-file-path")
+    @ConfigDescription("JSON key file used to access Google Cloud Storage")
+    public GcsFileSystemConfig setJsonKeyFilePath(String jsonKeyFilePath)
+    {
+        this.jsonKeyFilePath = jsonKeyFilePath;
+        return this;
+    }
+
+    public void validate()
+    {
+        // This cannot be normal validation, as it would make it impossible to write TestGcsFileSystemConfig.testExplicitPropertyMappings
```

Comment:
I'll try and update. This code and the comment were copied from HiveGcsConfig.

Reply:
Update: had to put the original code copied from HiveGcsConfig back, there is no way to exclude config properties from `testExplicitPropertyMappings` and one of `jsonKey` or `jsonKeyFilePath` must be null. 

One idea: can check for blank instead of null and then I can put it back. lmk if you want me to do that.